vpr/src/place/analytical_placement/AnalyticalSolver.h

```diff
@@ -33,7 +34,25 @@ class QPHybridSolver : public AnalyticalSolver {
     Eigen::SparseMatrix<double> A_sparse;
     Eigen::VectorXd b_x;
     Eigen::VectorXd b_y;
-
-    bool isSymmetric(const Eigen::SparseMatrix<double>&A);
-    bool isSemiPosDef(const Eigen::SparseMatrix<double>&A);
 };
+
+class B2BSolver : public AnalyticalSolver {
+    public:
+        void solve(unsigned iteration, PartialPlacement &p_placement) final;
+        void b2b_solve_loop(unsigned iteration, PartialPlacement &p_placement);
+        void initialize_placement_random_normal(PartialPlacement &p_placement);
+        void initialize_placement_random_uniform(PartialPlacement &p_placement);
+        void initialize_placement_least_dense(PartialPlacement &p_placement);
+        void populate_matrix(PartialPlacement &p_placement);
+        void populate_matrix_anchor(PartialPlacement& p_placement, unsigned iteration);
+        std::pair<size_t, size_t> boundNode(std::vector<size_t> &node_id, std::vector<double> &node_loc);
+
+        Eigen::SparseMatrix<double> A_sparse_x;
+        Eigen::SparseMatrix<double> A_sparse_y;
+        Eigen::VectorXd b_x;
+        Eigen::VectorXd b_y;
+        std::vector<double> node_loc_x_solved;
```

Comment:
Please add comments explaining the difference between these locs and why they are being stored.